app/models/derived_variables/sales_log_variables.rb

```diff
@@ -38,6 +38,10 @@ def set_derived_fields!
     self.nationality_all = nationality_all_group if nationality_uk_or_prefers_not_to_say?
     self.nationality_all_buyer2 = nationality_all_buyer2_group if nationality2_uk_or_prefers_not_to_say?
 
+    if saledate && form.start_year_after_2024? && discounted_ownership_sale?
+      self.ppostcode_full = postcode_full
```

Comment:
why are we checking saledate here? we already check `startdate` in `start_year_after_2024?`

Reply:
We check the `start_date` of the form not the log and the log might have the default form assigned before it has the date filled in